[BUG] Spark nightly cache tests fail with -- master flag

**Describe the bug**
When submitting cache tests in the integration tests [here](https://blossom.nvidia.com/sw-gpu-spark-jenkins/job/rapids_integration-dev-github/) with command 

```
spark-submit --master spark://raza-linux-2:7077 --executor-memory 12G --total-executor-cores 6 --conf spark.sql.shuffle.partitions=12 --conf spark.driver.extraClassPath=/home/rjafri/audit-plugin/jars/cudf-0.20-SNAPSHOT-cuda11.jar:/home/rjafri/audit-plugin/jars/rapids-4-spark_2.12-0.6.0-SNAPSHOT.jar:/home/rjafri/audit-plugin/jars/rapids-4-spark-udf-examples_2.12-0.6.0-SNAPSHOT.jar --conf spark.executor.extraClassPath=/home/rjafri/audit-plugin/jars/cudf-0.20-SNAPSHOT-cuda11.jar:/home/rjafri/audit-plugin/jars/rapids-4-spark_2.12-0.6.0-SNAPSHOT.jar:/home/rjafri/audit-plugin/jars/rapids-4-spark-udf-examples_2.12-0.6.0-SNAPSHOT.jar --conf spark.driver.extraJavaOptions=-Duser.timezone=UTC --conf spark.executor.extraJavaOptions=-Duser.timezone=UTC --conf spark.sql.session.timeZone=UTC --conf spark.sql.cache.serializer=com.nvidia.spark.rapids.shims.spark311.ParquetCachedBatchSerializer --jars /home/rjafri/audit-plugin/jars/rapids-4-spark-integration-tests_2.12-0.6.0-SNAPSHOT.jar ./runtests.py -v -rfExXs --std_input_path=/home/rjafri/audit-plugin/integration_tests/src/test/resources/ -k cache_test.py
```

They fail due to an exception. 

``` E                   py4j.protocol.Py4JJavaError: An error occurred while calling o282.collectToPython.
E                   : org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 4.0 failed 4 times, most recent failure: Lost task 0.3 in stage 4.0 (TID 33) (192.168.1.151 executor 0): java.lang.NullPointerException
E                   	at org.apache.spark.sql.internal.SQLConf.setConfWithCheck(SQLConf.scala:3856)
E                   	at org.apache.spark.sql.internal.SQLConf.setConfString(SQLConf.scala:3701)
E                   	at com.nvidia.spark.rapids.shims.spark311.ParquetCachedBatchSerializer$CachedBatchIteratorProducer$InternalRowToCachedBatchIterator.next(ParquetCachedBatchSerializer.scala:1081)
E                   	at com.nvidia.spark.rapids.shims.spark311.ParquetCachedBatchSerializer$CachedBatchIteratorProducer$InternalRowToCachedBatchIterator.next(ParquetCachedBatchSerializer.scala:948)
E                   	at scala.collection.Iterator$$anon$10.next(Iterator.scala:459)
E                   	at org.apache.spark.storage.memory.MemoryStore.putIterator(MemoryStore.scala:222)
E                   	at org.apache.spark.storage.memory.MemoryStore.putIteratorAsValues(MemoryStore.scala:299)
```
Removing the `-- master spark://aaa:bbb` flag from the command makes the script pass

**Steps/Code to reproduce bug**
set WORKSPACE to a local tmp folder and set `URM_URL=https://urm.nvidia.com/artifactory/sw-spark-maven`
run jenkins/spark-tests

**Expected behavior**
All tests should pass

--- a/shims/spark311/src/main/scala/com/nvidia/spark/rapids/shims/spark311/ParquetCachedBatchSerializer.scala
+++ b/shims/spark311/src/main/scala/com/nvidia/spark/rapids/shims/spark311/ParquetCachedBatchSerializer.scala
@@ -43,7 +43,7 @@ import org.apache.parquet.io.{DelegatingPositionOutputStream, DelegatingSeekable
 import org.apache.spark.TaskContext
 import org.apache.spark.broadcast.Broadcast
 import org.apache.spark.rdd.RDD
-import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.{vectorized, SparkSession}
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.catalyst.expressions.{Attribute, AttributeReference, Expression, GenericInternalRow, SpecializedGetters, UnsafeRow}
 import org.apache.spark.sql.catalyst.expressions.codegen.GenerateUnsafeProjection
@@ -609,11 +609,7 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
      sharedHadoopConf: Broadcast[SerializableConfiguration],
      sharedConf: Broadcast[Map[String, String]]) {
 
-    val conf = {
-      val c = new SQLConf()
-      sharedConf.value.foreach { case(k, v) => c.setConfString(k, v) }
-      c
-    }
+    val conf = getConfFromMap(sharedConf)
     val hadoopConf = sharedHadoopConf.value.value
     val origRequestedSchema: Seq[Attribute] = getCatalystSchema(selectedAttributes, cacheAttributes)
     val origCacheSchema: Seq[Attribute] = getCatalystSchema(cacheAttributes, cacheAttributes)
@@ -824,7 +820,7 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
         val columnVectors: Array[OffHeapColumnVector] =
           OffHeapColumnVector.allocateColumns(capacity, requestedSchema.toStructType)
         val columnarBatch = new ColumnarBatch(columnVectors
-          .asInstanceOf[Array[org.apache.spark.sql.vectorized.ColumnVector]])
+          .asInstanceOf[Array[vectorized.ColumnVector]])
         val missingColumns = new Array[Boolean](reqParquetSchema.getFieldCount)
         var columnReaders: Array[VectorizedColumnReader] = _
 
@@ -906,6 +902,12 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
     }
   }
 
+  private def getConfFromMap(sharedConf: Broadcast[Map[String, String]]): SQLConf = {
+    val conf = new SQLConf()
+    sharedConf.value.foreach { case (k, v) => conf.setConfString(k, v) }
+    conf
+  }
+
   private def getCatalystSchema(
      selectedAttributes: Seq[Attribute],
      cacheAttributes: Seq[Attribute]): Seq[Attribute] = {
@@ -937,11 +939,7 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
      sharedHadoopConf: Broadcast[SerializableConfiguration],
      sharedConf: Broadcast[Map[String, String]]) {
 
-    val conf = {
-      val c = new SQLConf()
-      sharedConf.value.foreach { case(k, v) => c.setConfString(k, v) }
-      c
-    }
+    val conf = getConfFromMap(sharedConf)
     val hadoopConf = sharedHadoopConf.value.value
 
     def getInternalRowToCachedBatchIterator: Iterator[CachedBatch] = {

--- a/shims/spark311/src/main/scala/com/nvidia/spark/rapids/shims/spark311/ParquetCachedBatchSerializer.scala
+++ b/shims/spark311/src/main/scala/com/nvidia/spark/rapids/shims/spark311/ParquetCachedBatchSerializer.scala
@@ -697,8 +697,7 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
           }
           parquetFileReader.close()
           val iter = unsafeRows.iterator
-          val unsafeProjection =
-            GenerateUnsafeProjection.generate(requestedSchema, cacheSchema)
+          val unsafeProjection = GenerateUnsafeProjection.generate(requestedSchema, cacheSchema)
           if (hasUnsupportedType) {
             new UnsupportedDataHandlerIterator() {
               val wrappedIter: Iterator[InternalRow] = iter

--- a/shims/spark311/src/main/scala/com/nvidia/spark/rapids/shims/spark311/ParquetCachedBatchSerializer.scala
+++ b/shims/spark311/src/main/scala/com/nvidia/spark/rapids/shims/spark311/ParquetCachedBatchSerializer.scala
@@ -327,11 +327,11 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
     } else {
       val cachedSchema = getCatalystSchema(schema, schema)
       val broadcastedHadoopConf = getBroadcastedHadoopConf(conf, cachedSchema)
-      val broadcastedConf = SparkSession.active.sparkContext.broadcast(conf)
+      val broadcastedConf = SparkSession.active.sparkContext.broadcast(conf.getAllConfs)
       input.mapPartitions {
         cbIter =>
           new CachedBatchIteratorProducer[ColumnarBatch](cbIter, cachedSchema,
-            broadcastedHadoopConf.value.value, broadcastedConf.value)
+            broadcastedHadoopConf, broadcastedConf)
             .getColumnarBatchToCachedBatchIterator
       }
     }
@@ -486,11 +486,11 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
       cbRdd.mapPartitions(iter => CloseableColumnBatchIterator(iter))
     } else {
       val broadcastedHadoopConf = getBroadcastedHadoopConf(conf, selectedAttributes)
-      val broadcastedConf = SparkSession.active.sparkContext.broadcast(conf)
+      val broadcastedConf = SparkSession.active.sparkContext.broadcast(conf.getAllConfs)
       input.mapPartitions {
         cbIter => {
           new CachedBatchIteratorConsumer(cbIter, cacheAttributes, selectedAttributes,
-            broadcastedHadoopConf.value.value, broadcastedConf.value).getColumnBatchIterator
+            broadcastedHadoopConf, broadcastedConf).getColumnBatchIterator
         }
       }
     }
@@ -518,11 +518,11 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
       }
     }
     val broadcastedHadoopConf = getBroadcastedHadoopConf(conf, selectedAttributes)
-    val broadcastedConf = SparkSession.active.sparkContext.broadcast(conf)
+    val broadcastedConf = SparkSession.active.sparkContext.broadcast(conf.getAllConfs)
     input.mapPartitions {
       cbIter => {
         new CachedBatchIteratorConsumer(cbIter, cacheAttributes, selectedAttributes,
-          broadcastedHadoopConf.value.value, broadcastedConf.value).getInternalRowIterator
+          broadcastedHadoopConf, broadcastedConf).getInternalRowIterator
       }
     }
   }
@@ -606,12 +606,18 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
      cbIter: Iterator[CachedBatch],
      cacheAttributes: Seq[Attribute],
      selectedAttributes: Seq[Attribute],
-     sharedHadoopConf: Configuration,
-     sharedConf: SQLConf) {
+     sharedHadoopConf: Broadcast[SerializableConfiguration],
+     sharedConf: Broadcast[Map[String, String]]) {
 
+    val conf = {
+      val c = new SQLConf()
+      sharedConf.value.foreach { case(k, v) => c.setConfString(k, v) }
+      c
+    }
+    val hadoopConf = sharedHadoopConf.value.value
     val origRequestedSchema: Seq[Attribute] = getCatalystSchema(selectedAttributes, cacheAttributes)
     val origCacheSchema: Seq[Attribute] = getCatalystSchema(cacheAttributes, cacheAttributes)
-    val options: ParquetReadOptions = HadoopReadOptions.builder(sharedHadoopConf).build()
+    val options: ParquetReadOptions = HadoopReadOptions.builder(hadoopConf).build()
     /**
      * We are getting this method using reflection because its a package-private
      */
@@ -681,7 +687,7 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
             val recordReader =
               columnIO.getRecordReader(pages, new ParquetRecordMaterializer(parquetSchema,
                 cacheSchema.toStructType,
-                new ParquetToSparkSchemaConverter(sharedHadoopConf), None /*convertTz*/,
+                new ParquetToSparkSchemaConverter(hadoopConf), None /*convertTz*/,
                 LegacyBehaviorPolicy.CORRECTED))
             for (_ <- 0 until rows.toInt) {
               val row = recordReader.read
@@ -781,7 +787,7 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
      */
     def getColumnBatchIterator: Iterator[ColumnarBatch] = {
       if (!cbIter.hasNext) return Iterator.empty
-      val capacity = sharedConf.parquetVectorizedReaderBatchSize
+      val capacity = conf.parquetVectorizedReaderBatchSize
       val parquetCachedBatch = cbIter.next().asInstanceOf[ParquetCachedBatch]
       val inputFile = new ByteArrayInputFile(parquetCachedBatch.buffer)
       val parquetFileReader = ParquetFileReader.open(inputFile, options)
@@ -794,9 +800,9 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
       // we are getting parquet schema and then converting it to catalyst schema
       // because catalyst schema that we get from Spark doesn't have the exact schema expected by
       // the columnar parquet reader
-      val parquetToSparkSchemaConverter = new ParquetToSparkSchemaConverter(sharedHadoopConf)
+      val parquetToSparkSchemaConverter = new ParquetToSparkSchemaConverter(hadoopConf)
       val sparkSchema = parquetToSparkSchemaConverter.convert(parquetSchema)
-      val sparkToParquetSchemaConverter = new SparkToParquetSchemaConverter(sharedHadoopConf)
+      val sparkToParquetSchemaConverter = new SparkToParquetSchemaConverter(hadoopConf)
 
       val (_, requestedSchema) = if (hasUnsupportedType) {
         getSupportedSchemaFromUnsupported(origCacheSchema, origRequestedSchema)
@@ -809,8 +815,8 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
       val reqParquetSchema = sparkToParquetSchemaConverter.convert(reqSparkSchema)
 
       // reset spark schema calculated from parquet schema
-      sharedHadoopConf.set(ParquetReadSupport.SPARK_ROW_REQUESTED_SCHEMA, reqSparkSchema.json)
-      sharedHadoopConf.set(ParquetWriteSupport.SPARK_ROW_SCHEMA, reqSparkSchema.json)
+      hadoopConf.set(ParquetReadSupport.SPARK_ROW_REQUESTED_SCHEMA, reqSparkSchema.json)
+      hadoopConf.set(ParquetWriteSupport.SPARK_ROW_SCHEMA, reqSparkSchema.json)
 
       /**
        * Read the next RowGroup and read each column and return the columnarBatch
@@ -929,8 +935,15 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
   private[rapids] class CachedBatchIteratorProducer[T](
      iter: Iterator[T],
      cachedAttributes: Seq[Attribute],
-     sharedHadoopConf: Configuration,
-     sharedConf: SQLConf) {
+     sharedHadoopConf: Broadcast[SerializableConfiguration],
+     sharedConf: Broadcast[Map[String, String]]) {
+
+    val conf = {
+      val c = new SQLConf()
+      sharedConf.value.foreach { case(k, v) => c.setConfString(k, v) }
+      c
+    }
+    val hadoopConf = sharedHadoopConf.value.value
 
     def getInternalRowToCachedBatchIterator: Iterator[CachedBatch] = {
       new InternalRowToCachedBatchIterator
@@ -965,9 +978,9 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
             getSupportedSchemaFromUnsupported(getCatalystSchema(cachedAttributes,
               cachedAttributes))._1
           // save it to sharedConf and sharedHadoopConf
-          sharedHadoopConf.set(
+          hadoopConf.set(
             ParquetReadSupport.SPARK_ROW_REQUESTED_SCHEMA, newCachedAttributes.toStructType.json)
-          sharedHadoopConf.set(
+          hadoopConf.set(
             ParquetWriteSupport.SPARK_ROW_SCHEMA, newCachedAttributes.toStructType.json)
           newCachedAttributes
         } else {
@@ -1077,10 +1090,10 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
             // at least a single block
             val stream = new ByteArrayOutputStream(ByteArrayOutputFile.BLOCK_SIZE)
             val outputFile: OutputFile = new ByteArrayOutputFile(stream)
-            sharedConf.setConfString(ShimLoader.getSparkShims.parquetRebaseWriteKey,
+            conf.setConfString(ShimLoader.getSparkShims.parquetRebaseWriteKey,
               LegacyBehaviorPolicy.CORRECTED.toString)
-            val recordWriter = SQLConf.withExistingConf(sharedConf) {
-              parquetOutputFileFormat.getRecordWriter(outputFile, sharedHadoopConf)
+            val recordWriter = SQLConf.withExistingConf(conf) {
+              parquetOutputFileFormat.getRecordWriter(outputFile, hadoopConf)
             }
             var totalSize = 0
             while (rowIterator.hasNext && totalSize < BYTES_ALLOWED_PER_BATCH) {
@@ -1264,12 +1277,11 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
       })
     } else {
       val broadcastedHadoopConf = getBroadcastedHadoopConf(conf, parquetSchema)
-      val broadcastedConf = SparkSession.active.sparkContext.broadcast(conf)
-      // fallback to the CPU
+      val broadcastedConf = SparkSession.active.sparkContext.broadcast(conf.getAllConfs)
       input.mapPartitions {
         cbIter =>
           new CachedBatchIteratorProducer[InternalRow](cbIter, schema,
-            broadcastedHadoopConf.value.value, broadcastedConf.value)
+            broadcastedHadoopConf, broadcastedConf)
             .getInternalRowToCachedBatchIterator
       }
     }

--- a/tests-spark310+/src/test/scala/com/nvidia/spark/rapids/Spark310ParquetWriterSuite.scala
+++ b/tests-spark310+/src/test/scala/com/nvidia/spark/rapids/Spark310ParquetWriterSuite.scala
@@ -26,11 +26,13 @@ import org.mockito.ArgumentMatchers._
 import org.mockito.Mockito._
 import org.mockito.invocation.InvocationOnMock
 
+import org.apache.spark.SparkContext
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.catalyst.expressions.AttributeReference
 import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.types._
 import org.apache.spark.sql.vectorized.ColumnarBatch
+import org.apache.spark.util.SerializableConfiguration
 
 
 /**
@@ -167,8 +169,10 @@ class Spark310ParquetWriterSuite extends SparkQueryCompareTestSuite {
       }
     }
     val ser = new ParquetCachedBatchSerializer
+
     val producer = new ser.CachedBatchIteratorProducer[ColumnarBatch](cbIter, schema,
-      new Configuration(true), new SQLConf)
+      SparkContext.getOrCreate().broadcast(new SerializableConfiguration(new Configuration(true))),
+      SparkContext.getOrCreate().broadcast(new SQLConf().getAllConfs))
     val mockParquetOutputFileFormat = mock(classOf[ParquetOutputFileFormat])
     var totalSize = 0L
     val mockRecordWriter = new RecordWriter[Void, InternalRow] {

--- a/tests-spark310+/src/test/scala/com/nvidia/spark/rapids/Spark310ParquetWriterSuite.scala
+++ b/tests-spark310+/src/test/scala/com/nvidia/spark/rapids/Spark310ParquetWriterSuite.scala
@@ -171,8 +171,9 @@ class Spark310ParquetWriterSuite extends SparkQueryCompareTestSuite {
     val ser = new ParquetCachedBatchSerializer
 
     val producer = new ser.CachedBatchIteratorProducer[ColumnarBatch](cbIter, schema,
-      SparkContext.getOrCreate().broadcast(new SerializableConfiguration(new Configuration(true))),
-      SparkContext.getOrCreate().broadcast(new SQLConf().getAllConfs))
+      withCpuSparkSession(spark =>
+        spark.sparkContext.broadcast(new SerializableConfiguration(new Configuration(true)))),
+      withCpuSparkSession(spark => spark.sparkContext.broadcast(new SQLConf().getAllConfs)))
     val mockParquetOutputFileFormat = mock(classOf[ParquetOutputFileFormat])
     var totalSize = 0L
     val mockRecordWriter = new RecordWriter[Void, InternalRow] {

